src/chainparams.cpp

```diff
@@ -334,9 +335,15 @@ CChainParams* CChainParams::Factory(CBaseChainParams::Network network, CScript s
     }
 }
 
+CChainParams &Params(CBaseChainParams::Network network)
+{
+    globalSwitchingChainParams.reset(CChainParams::Factory(network, CScript()));
+    return *globalSwitchingChainParams;
+}
+
```

Comment:
I considered this, and decided it felt very weird for `Params()` to be `const` while `Params(network)` has global side-effects.

I think my PR is functionally identical.